subsys/usb/device/usb_device.c

```diff
@@ -726,8 +811,18 @@ static bool usb_set_interface(struct usb_setup_packet *setup)
 		case USB_DESC_ENDPOINT:
 			if (cur_iface == setup->wIndex) {
 				ep = (struct usb_ep_descriptor *)p;
-				ret = usb_eps_reconfigure(ep, cur_alt_setting,
-							  setup->wValue);
+
+				/*
+				 * only reconfigure endpoint if either:
+				 * a) currently iterating over selected alt_setting
+				 * b) endpoint not used in selected alt_setting
+				 */
+				ep_bm = get_ep_bm_from_addr(ep->bEndpointAddress);
+				if (cur_alt_setting == setup->wValue ||
+				    (ep_bm_active & ep_bm) == 0) {
+					ret = usb_eps_reconfigure(
```

Comment:
Should there be also a bitmap of endpoints already reconfigured? If there are three alternate settings, the `usb_eps_reconfigure()` can be called twice on endpoint not used in selected alternate setting.

Reply:
Good point - thanks - added